hazelcast/src/test/java/com/hazelcast/client/alto/ClientAltoTest.java

```diff
@@ -338,6 +383,71 @@ public void testAltoDisabledClient_inAltoEnabledCluster() {
         assertEquals("42", map.get("42"));
     }
 
+    @Test
+    public void testAltoClient_heartbeatsToIdleAltoChannels() {
+        Config config = getMemberConfig();
+        Hazelcast.newHazelcastInstance(config);
+
+        ClientConfig clientConfig = getClientConfig();
+        clientConfig.setProperty(HEARTBEAT_INTERVAL.getName(), "1000");
+
+        HazelcastInstance client = HazelcastClient.newHazelcastClient(clientConfig);
+
+        ClientConnectionManager connectionManager = getConnectionManager(client);
+        Collection<ClientConnection> connections = connectionManager.getActiveConnections();
+        assertClientConnectsAllAltoPortsEventually(connections, config.getAltoConfig().getEventloopCount());
+
+        ClientConnection connection = connectionManager.getRandomConnection();
+        assertTrue(connection.isAlive());
+
+        Channel[] altoChannels = connection.getAltoChannels();
+        assertNotNull(altoChannels);
+
+        long now = System.currentTimeMillis();
+        assertTrueEventually(() -> {
+            for (Channel channel : altoChannels) {
+                assertTrue(channel.lastWriteTimeMillis() > now);
+            }
+        });
```

Comment:
Even with 2 seconds, It would be too time sensitive, and would certainly fail a lot. The exact time in which this will be executed is dependent on the executor we are using, so I would say this is not related to this PR. Making sure that we sent heartbeats is enough IMO